### PR TITLE
Update README and examples to reflect v1 changes

### PR DESCRIPTION
**Pull request by aulorbe**
## Problem

We are close to launching v1! With it, our README needs a facelift. Here is that facelift 💅. 
We have also updated the SDK version, user-agent string, and the examples package.

Asana tix: 
- Audrey: https://app.asana.com/0/1203260648987893/1207042384514994/f
- Rohan: https://app.asana.com/0/1203260648987893/1206745140276838/f

## Solution

Update README, user-agent string, and examples for v1 java SDK. We have currently removed the UpsertsAndQueriesConcurrentExample example but in upcoming PR's, we will like to add it back.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [x] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,35 +8,467 @@ The Pinecone Java client lets JVM applications interact with Pinecone services.
 
 ## Installation
 
-*pinecone-client* can be installed from Maven Central for use as a maven dependency in the following ways:
+*pinecone-client* can be installed from [Maven Central](https://mvnrepository.com/repos/central) for use as a maven dependency in the following ways:
```
Comment (ssmith-pc) [inline README.md:11]: I would link to the top level artifact directly in maven repo: 

`https://mvnrepository.com/artifact/io.pinecone/pinecone-client`

File: README.md [hunk 2]
```diff
@@ -8,35 +8,467 @@ The Pinecone Java client lets JVM applications interact with Pinecone services.
...
+        OkHttpClient.Builder builder = new OkHttpClient.Builder()
+                .connectTimeout(10, java.util.concurrent.TimeUnit.SECONDS)
+                .readTimeout(30, java.util.concurrent.TimeUnit.SECONDS)
+                .writeTimeout(30, java.util.concurrent.TimeUnit.SECONDS);
+
+        OkHttpClient httpClient = builder.build();
+
+        Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();
+    }
+}
```
Comment (ssmith-pc) [inline README.md:80]: For sake of brevity, I think we can exclude the Java main wrapper and import statements from all but the first of these examples.

The exception for imports would be if we're importing some various Pinecone classes besides the Pinecone client, but we can still exclude the java main wrapper in those cases.

E.g.: 
```java      
Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();
        
String indexName = "example-index";
String similarityMetric = "cosine";
Integer dimension = 1538;
String cloud = "aws";
String region = "us-west-2";

pinecone.createServerlessIndex(indexName, similarityMetric, dimension, cloud, region);
```

and 

```java
import io.pinecone.clients.Index;
import io.pinecone.clients.Pinecone;
...

Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();

Index index = pinecone.getIndexConnection("example-index");
DescribeIndexStatsResponse indexStatsResponse = index.describeIndexStats();
```

File: README.md [hunk 3]
```diff
@@ -8,35 +8,467 @@ The Pinecone Java client lets JVM applications interact with Pinecone services.
...
+import org.openapitools.client.model.*;
+
+public class ListIndexExample {
+    public static void main(String[] args) {
+        Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();
+        
+        IndexList indexesInYourProject = pinecone.listIndexes();
+        
+        System.out.println(indexesInYourProject);
+    }
```
Comment (ssmith-pc) [inline README.md:153]: When these methods return something I like how you assigned it so the return type is visible. 

We can probably exclude the System.out.println statements

File: README.md [hunk 4]
```diff
@@ -8,35 +8,467 @@ The Pinecone Java client lets JVM applications interact with Pinecone services.
...
+your `apiKey`, either by passing it as an argument in your code or by setting it as an environment variable called 
+`PINECONE_API_KEY`.
+
+Note: for pod-based indexes, you will also need an `environment` variable. You can set pass this as an argument in 
+your code or set it as an environment variable called `PINECONE_ENVIRONMENT`.
+
+```java
+import io.pinecone.clients.Pinecone;
+import org.openapitools.client.model.*;
+
```
Comment (ssmith-pc) [inline README.md:57]: I see this openapitools imported everywhere, do we need it?

File: README.md (hunk 3 above)
Comment (austin-denoble) [reply README.md:153]: I followed this pattern as well in my PR. https://github.com/pinecone-io/pinecone-java-client/pull/109

File: README.md (hunk 4 above)
Comment (austin-denoble) [reply README.md:57]: We do for now, it's where all our generated code for the control plane lives, so people need to import the models from there.

Comment (rohanshah18) [reply README.md:57]: Yeah we dont need to import all of the org.openapitools.client.model.*, so I have replaced it with the specific class we need to import it with

File: README.md (hunk 2 above)
Comment (rohanshah18) [reply README.md:80]: Done!

File: src/main/java/io/pinecone/configs/PineconeConfig.java [hunk 5]
```diff
@@ -176,7 +176,7 @@ public String getUserAgentGrpc() {
     }
 
     private String buildUserAgent(String clientId) {
-        String userAgent = String.format("lang=java; %s=%s", clientId, "v0.8.0");
+        String userAgent = String.format("lang=java; %s=%s", clientId, "v1.0.0");
```
Comment (rohanshah18) [inline src/main/java/io/pinecone/configs/PineconeConfig.java:179]: In the future PR, I'm going to automate this to extract version from build.gradle or some other file so we only have to update this version at one place

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,12 @@
 
 [comment]: <> (When bumping [pc:VERSION_LATEST_RELEASE] create a new entry below)
 ### Unreleased version
+### v0.1.0
```
Comment (austin-denoble) [inline CHANGELOG.md:5]: Should this be `v1.0.0`?

File: examples/java-basic-mvn/src/main/java/pineconeexamples/MinimalUpsertAndQueryExample.java [hunk 7]
```diff
@@ -1,93 +1,43 @@
...
-                    .build();
-
-            System.out.println("Sending upsert request:");
-            System.out.println(upsertRequest);
-
-            UpsertResponse upsertResponse = connection.getBlockingStub().upsert(upsertRequest);
-
-            System.out.println("Got upsert response:");
-            System.out.println(upsertResponse);
+        Pinecone pinecone = new Pinecone.Builder(args.apiKey).build();
```
Comment (austin-denoble) [inline examples/java-basic-mvn/src/main/java/pineconeexamples/MinimalUpsertAndQueryExample.java:26]: Much cleaner with the new architecture, thanks for updating!

**Review by austin-denoble — APPROVED** [review]
Overall looks good, I'm not sure if the version comment in the `CHANGELOG` file needs to be updated to `v1.0.0` though.

File: CHANGELOG.md (hunk 6 above)
Comment (rohanshah18) [reply CHANGELOG.md:5]: YES!!! Let me update it, thanks!

Comment (rohanshah18) [reply CHANGELOG.md]: Done!

**Pull request merged**